src/build/mdn-comments.ts

```
@@ -47,66 +45,124 @@ function extractSummary(markdown: string): string {
+  } else {
+    const [ifaceName, memberName] = slug;
+    const iface = ensureLeaf(root, [ifaceName, ...types]);
+
+    const mainComment = root[ifaceName]?.__comment;
+    if (mainComment !== summary) {
+      iface[memberName] = { comment: summary };
+    }
+  }
 }
 
-export async function generateDescription(): Promise<Record<string, string>> {
+export async function generateDescriptions(removedComments: string[]): Promise<{
+  interfaces: { interface: Record<string, any> };
+}> {
   const stats = await fs.stat(basePath);
   if (!stats.isDirectory()) {
     throw new Error(
       "MDN submodule does not exist; try running `git submodule update --init`",
     );
   }
+
+  const results: Record<string, any> = {};
   try {
-    const folders = await getDirectories(basePath);
-    if (folders.length > 0) {
-      return await getIndexMdContents(folders);
-    }
-  } catch (error) {
-    console.error("Error generating API descriptions:", error);
-  }
+    const indexPaths = await walkDirectory(basePath);
+
+    await Promise.all(
+      indexPaths.map(async (fileURL) => {
+        try {
+          const content = await fs.readFile(fileURL, "utf-8");
+          const slug = generateSlug(content);
+          const types = generateTypes(content);
+          if (
+            !slug ||
+            slug.length === 0 ||
+            removedComments.includes(slug[0]) ||
+            (types && types[0] === "ignore")
+          )
+            return;
 
-  return {};
+          const summary = extractSummary(content);
+          if (
+            summary ===
+            "When writing code for the Web, there are a large number of Web APIs available."
+          )
+            console.log(types, slug, summary);
+          insertComment(results, slug, summary, types);
+        } catch (error) {
+          console.error("Error generating API descriptions:", error);
+        }
+      }),
+    );
+  } catch (err) {
+    console.error("Error generating API descriptions:", err);
```

Please remove any try-catch here and let the code throw.